champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/passives/Adrenaline.java

```diff
@@ -0,0 +1,122 @@
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.scheduler.BukkitTask;
+
+import java.util.WeakHashMap;
+
+@Singleton
+@BPvPListener
+public class Adrenaline extends Skill implements PassiveSkill, Listener, BuffSkill, OffensiveSkill {
+
+    private double speedOneHealth;
+    private double speedOneHealthIncreasePerLevel;
+    private double speedTwoHealth;
+    private double speedTwoHealthIncreasePerLevel;
+
+    private final WeakHashMap<Player, Integer> playerSpeedLevels = new WeakHashMap<>();
+
+    @Inject
+    public Adrenaline(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Adrenaline";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
+                "Below " + getValueString(this::getSpeedOneHealth, level, 100, "%", 1) + " health you gain <effect>Speed I</effect>,",
+                "and below " + getValueString(this::getSpeedTwoHealth, level, 100, "%", 1) + " health you gain <effect>Speed II</effect> ",
+        };
+    }
+
+    public double getSpeedOneHealth(int level) {
+        return speedOneHealth + ((level - 1) * speedOneHealthIncreasePerLevel);
+    }
+
+    public double getSpeedTwoHealth(int level) {
+        return speedTwoHealth + ((level - 1) * speedTwoHealthIncreasePerLevel);
+    }
+
+    @Override
+    public Role getClassType() {
+        return Role.BRUTE;
+    }
+
+    @Override
+    public SkillType getType() {
+        return SkillType.PASSIVE_B;
+    }
+
+    @UpdateEvent
+    public void giveSpeed() {
+        for (Player player : playerSpeedLevels.keySet()) {
+            int level = getLevel(player);
+            if (level > 0) {
+                double healthThresholdSpeedOne = getSpeedOneHealth(level) * player.getMaxHealth();
+                double healthThresholdSpeedTwo = getSpeedTwoHealth(level) * player.getMaxHealth();
+
+                if (player.getHealth() <= healthThresholdSpeedTwo) {
+                    // Player should have Speed II
+                    championsManager.getEffects().addEffect(player, player, EffectTypes.SPEED, 2, 500);
+                    playerSpeedLevels.put(player, 2);
+                } else if (player.getHealth() <= healthThresholdSpeedOne) {
+                    // Player should have Speed I
+                    championsManager.getEffects().addEffect(player, player, EffectTypes.SPEED, 1, 500);
+                    playerSpeedLevels.put(player, 1);
+                } else {
+                    Integer currentSpeedLevel = playerSpeedLevels.get(player);
+                    if (currentSpeedLevel != null && currentSpeedLevel > 0) {
+                        championsManager.getEffects().removeEffect(player, EffectTypes.SPEED);
+                        playerSpeedLevels.put(player, 0);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
A few changes needed

- In addEffect, add the name of the effect (e.g. https://github.com/Mykindos/BetterPvP/blob/c5e8793925eea427c7a8c679955268714ff77fbe/champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/passives/Bloodlust.java#L106-L107)

- In removeEffect, add the same name as the last argument, this way we only remove speed provided by the passive and not speed that might come from other buffs

- To keep this as balanced as possible, just give speed for 100ms instead of 500, we dont need it running longer than intended if a player regenerates above the threshold

- I dont think this needs to run every tick, maybe every 2 ticks (can increase the duration to 150 if worried about it expiring and causing FoV issues)

- You aren't actually using the value in the playerSpeedLevels map, suggest just making this a List (either of WeakReference<Player> or just UUID)